### PR TITLE
Use Guild member counts rather than user cache size

### PR DESCRIPTION
**Pull request by Lioness100**
*This is mostly a fix for discord.js v12+*

Currently, `DiscordJS#userCount` relies on the user cache, which will result in inaccurate numbers. (Much) more often than not, the user cache will not include every single user. Instead, it would be better to use the guild cache (since by default they're all guaranteed to be there) using `Collection#reduce()` and `Guild#memberCount`

### CODE REVIEW

**Comment by EndBug** [comment]
Hi, thanks for the PR!
At the moment I'm waiting to marge a PR that changes how the lib folder is handled, so we'll need to update this PR after I merge that.

Anyway, regarding your contribution: the change looks good to me. The tests for `Poster#post` are failing, the dummy client should be updated to fit the new way of getting the user count.

**Comment by Snazzah** [comment]
This change should be mirrored to the other client fillers if possible. I'm pretty sure the other client fillers rely on user cache rather than member count.

**Review by EndBug — APPROVED** [review]
@Snazzah Maybe it's better if we merge this PR and change only discord.js for now. 
Then we can always change the other ones too, but maybe it's better if you do it, I'm not really that familiar with the other libraries.

**Comment by Lioness100** [comment]
This looks great!

File: src/Interface/Clients/DiscordIO.ts [hunk 1]
```diff
@@ -7,8 +8,13 @@ import { ClientFiller } from '../ClientFiller'
  */
 export default class DiscordIO extends ClientFiller {
   get userCount() {
-    if (!this.client.users) return undefined
-    return Object.keys(this.client.users).length
+    if (!this.client.servers) return undefined
+    return Object.keys(this.client.servers)
+      .map((id) => this.client.servers[id])
```
Comment (EndBug) [inline src/Interface/Clients/DiscordIO.ts:13]: I think we can just use `Object.values`

```suggestion
    return Object.values(this.client.servers)
```

**Review by EndBug — APPROVED** [review]
Awesome!

**Review by Snazzah — APPROVED** [review]

**Comment by EndBug** [comment]
@Lioness100 All set then! Thanks again for your contribution! The fix will be published in the next version 👍🏻

**Pull request merged**